pytd/writer.py

```diff
@@ -2,36 +2,309 @@
 import logging
 import os
 import re
+import tempfile
+import time
 from urllib.error import HTTPError
 from urllib.request import urlopen
 
+TD_SPARK_BASE_URL = "https://s3.amazonaws.com/td-spark/{}"
+TD_SPARK_JAR_NAME = "td-spark-assembly_2.11-19.7.0.jar"
 logger = logging.getLogger(__name__)
 
-TD_SPARK_BASE_URL = "https://s3.amazonaws.com/td-spark/%s"
-TD_SPARK_JAR_NAME = "td-spark-assembly_2.11-1.1.0.jar"
+
+def _cast_dtypes(dataframe):
+    """Convert dtypes into one of {int, float, str} type.
+
+    A character code (one of ‘biufcmMOSUV’) identifying the general kind of data.
+
+    b  boolean
+    i  signed integer
+    u  unsigned integer
+    f  floating-point
+    c  complex floating-point
+    m  timedelta
+    M  datetime
+    O  object
+    S  (byte-)string
+    U  Unicode
+    V  void
+    """
+    df = dataframe.copy()
```

Comment:
Just confirming. Does this line mean copy for every conversion? Don't we need some option like `inplace?`

Reply:
Agreed. eb966de58698d1d7cbcadb7aec7cfd99d586116e